lib/matplotlib/axis.py

```diff
@@ -1784,7 +1785,11 @@ def set_pickradius(self, pickradius):
         ----------
         pickradius :  float
         """
-        self.pickradius = pickradius
+        if not isinstance(pickradius, Number) or pickradius < 0:
+            raise ValueError("pick radius should be a distance")
+        self._pickradius = pickradius
+
+    pickradius = property(get_pickradius, set_pickradius)
```

Comment:
It seems that sphinx is recognizing attributes that are actually properties and tries to use that. In particular, it tries to link to the property. See e.g. the `axes` parameter above https://output.circle-artifacts.com/output/job/e39ebdcd-444d-4cea-8357-c5789279aa1e/artifacts/0/doc/build/html/api/axis_api.html?highlight=axis#matplotlib.axis.Axis

I suggest to try and add a `doc=...` value here. That should prevent sphinx trying to get the documentation from the getter.
More importantly, I suspect that autodoc only creates the documentation page for the property if it has documentation, and that could solve your link problem.

Reply:
Thanks! I gave it a go. Let's see what happens...